mlflow/tracking/_model_registry/client.py

```diff
@@ -30,7 +30,10 @@ def __init__(self, registry_uri):
         :param registry_uri: Address of local or remote model registry server.
         """
         self.registry_uri = registry_uri
-        self.store = utils._get_store(self.registry_uri)
+
+    @property
+    def store(self):
+        return utils._get_store(self.registry_uri)
```

Comment:
nice trick!

Reply:
Now the store object is lazily initialized. It seems this behavior change broke some tests:

https://github.com/mlflow/mlflow/pull/4719/checks?check_run_id=3379318839#step:4:106

Reply:
Great suggestion! I've implemented it to preserve existing behavior, which provides useful up-front validation outside of tests too.